### PR TITLE
Redis retry

### PR DESCRIPTION
**Pull request by allanchau**
This PR updates Kue to use ioredis instead of node_redis (default).

## Related PRs

- #52 

## Testing

- [x] Add this code to a remote cache deployment and look out for reconnect logs.
- [x] Make sure the new redis test passes.

## Notes

- This is currently on the AB beta sites.

### CODE REVIEW

**Comment by allanchau** [comment]
@smebberson I've updated it to use ioredis now, but left the clusters out. Perhaps that should be another PR to make things simpler?

File: idearium-lib/common/redis.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const config = require('./config');
+const Redis = require('ioredis');
+const log = require('./log')('idearium-lib:common/redis');
+
+Redis.Promise.onPossiblyUnhandledRejection(err => log.error({ err }, 'Unhandled Redis error'));
```
Comment (smebberson) [inline idearium-lib/common/redis.js]: @allanchau, what is this and how does it work?

File: idearium-lib/common/redis.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        // 10 x 2000ms is about 2 minutes
+        const retryLimit = config.get('redisRetryLimit') || 10;
+        const retryDelay = config.get('redisRetryDelay') || 2000;
+
+        const redis = new Redis(config.get('cacheUrl'), {
+            retryStrategy: (times) => {
+
+                if (times >= retryLimit) {
+
+                    log.fatal('Retry limit reached, could not connect to Redis');
```
Comment (smebberson) [inline idearium-lib/common/redis.js]: Could we please put the `redisRetryLimit` value in the message here...

Comment (allanchau) [reply idearium-lib/common/redis.js]: @smebberson Updated now.

**Pull request merged**